docs: add rfc template (#226)
razorpay/blade: rfcs/0000-template.md

---
Start Date: DD-MM-YYYY
RFC PR: (leave this empty if no PR yet)
Blade Issue: (leave this empty if no issue yet)
---

# Title of the RFC <!-- omit in toc -->

## Table Of Contents <!-- omit in toc -->
- [Summary](#summary)
- [Basic Example](#basic-example)
- [Motivation](#motivation)
- [Detailed Design](#detailed-design)
- [Drawbacks/Constraints](#drawbacksconstraints)
- [Alternatives](#alternatives)
- [Adoption strategy](#adoption-strategy)
- [How Do We Educate People?](#how-do-we-educate-people)
- [Open Questions](#open-questions)
- [References](#references)

# Summary
Brief explanation about the proposal.

# Basic Example
Include a basic code example. Omit this section if it's not applicable.

# Motivation
Why are we doing this? What use cases does it support? What is the expected outcome?

> Please focus on explaining the motivation so that if this RFC is not accepted, the motivation could be used to develop alternative solutions. In other words, enumerate the constraints you are trying to solve without coupling them too closely to the solution you have in mind.

Reviewer: ```suggestion
Try to focus on explaining the motivation so that if this RFC is not accepted, the motivation could be used to develop alternative solutions. In other words, enumerate the constraints you are trying to solve without coupling them too closely to the solution you have in mind.
```

Reviewer: "enumerate the constraints you are trying to solve without coupling them too closely to the solution you have in mind."

Maybe some easier words for this? Seems too technical

# Detailed Design
This is the bulk of the RFC. Explain the design in enough detail for somebody familiar with the Design System to understand, and for somebody familiar with the implementation to implement. This should get into specifics and corner-cases, and include examples of how the feature is used. Any new terminology should be defined here.

# Drawbacks/Constraints

> Why should we *not* do this? Please consider:
> 
> - Implementation cost, both in terms of code size and complexity.
> - The impact of it on new as well as existing consumer projects.
> - Cost of migration.

Reviewer: The tone of the words might seem a little demeaning

```suggestion
Why should we *not* do this?

Consider the following constraints:
- Implementation cost, both in terms of code size and complexity.
- The impact of it on new as well as existing consumer projects.
- Cost of migration.
```

There are tradeoffs to choosing any path. Attempt to identify them here.

# Alternatives
What other designs/patterns/strategies have been considered?

# Adoption strategy
If we implement this proposal, how will existing consumer projects adopt it? 
- Is this a breaking change? 
- Can we write a codemod?
- How do we prioritise this with business and product folks?
- How do we communicate with other teams? Will updating docs suffice or do we need a dedicated interaction with them?

> # How Do We Educate People?

Reviewer: Do we need to capitalize this?
```suggestion
# How do we educate people?
```

---
Start Date: DD-MM-YYYY
RFC PR: (leave this empty if no PR yet)
Blade Issue: (leave this empty if no issue yet)
---

# Title of the RFC <!-- omit in toc -->

### Table Of Contents <!-- omit in toc -->
- [Summary](#summary)
- [Basic Example](#basic-example)
- [Motivation](#motivation)
- [Detailed Design](#detailed-design)
- [Drawbacks/Constraints](#drawbacksconstraints)
- [Alternatives](#alternatives)
- [Adoption strategy](#adoption-strategy)
- [How do we educate people?](#how-do-we-educate-people)
- [Open Questions](#open-questions)
- [References](#references)

> # Summary

Reviewer 2: make this and below headers as h2 ?

Author: the `h1` are intentional

Brief explanation about the proposal.

# Basic Example
Include a basic code example. Omit this section if it's not applicable.

# Motivation
- Why are we doing this?
- What use cases does it support?
- What is the expected outcome?

> Try to focus on explaining the motivation so that if this RFC is not accepted, the motivation could be used to develop alternative solutions. In other words, try to list down the constraints you are trying to solve without coupling them too closely to the solution you have in mind.

Reviewer 2: Is this section about the solution or the motivation? Since it says:
> try to list down the constraints you are trying to solve without coupling them too closely to the solution you have in mind.

and then here it says
> What use cases does it support?

Seems confusing to me.

Reviewer 3: This section is about motivation of why do we even need to do this? What is this solving? It's not about how to solution, it's about the end result and what it would solve.

Author: So the section says try to talk about motivation and philosophies so even if the RFC is not accepted we can still use some ideas from this section. Hence it says
>In other words, try to list down the constraints you are trying to solve without coupling them too closely to the solution you have in mind.

# Detailed Design
This is the bulk of the RFC. Explain the design in enough detail for somebody familiar with the Design System to understand, and for somebody familiar with the implementation to implement. This should get into specifics and corner-cases, and include examples of how the feature is used. Any new terminology should be defined here.

# Drawbacks/Constraints
Why should we *not* do this? Maybe try to consider the following constraints

> - Implementation cost, both in terms of code size and complexity.

Reviewer 2: what does code size mean? Man hours/Time required?

Reviewer 3: Not sure if the RFC authors would be able to give man hours at time of RFC.
Probably, we can provide a time range based size. Like:
- Small (0-5 days or 1 week)
- Medium (5-10 days or 2 weeks)
- Large (More than 10 days or more than 2 weeks)
- XLarge (More than 20 days or more than 4 weeks)

Author: This doesn't asks you to give time estimates. It just tells you to list down constraints taking these perspectives in mind
